homeassistant/helpers/translation.py

```diff
@@ -465,20 +470,35 @@ def async_setup(hass: HomeAssistant) -> None:
 
     Listeners load translations for every loaded component and after config change.
     """
+    cache = _TranslationCache(hass)
 
-    hass.data[TRANSLATION_FLATTEN_CACHE] = _TranslationCache(hass)
+    hass.data[TRANSLATION_FLATTEN_CACHE] = cache
 
-    async def load_translations(event: Event) -> None:
-        if "language" in event.data:
-            language = hass.config.language
-            _LOGGER.debug("Loading translations for language: %s", language)
-            await _async_load_state_translations_to_cache(hass, language, None)
+    @callback
+    def _async_load_translations_filter(event: Event) -> bool:
+        """Filter out unwanted events."""
+        return "language" in event.data
```

Comment:
I think this should be smarter, there's no guarantee the language has *changed* just because we get this event, I think we need to store the old language.

Reply:
adjusted in 4894d8f3e586556db825cbf5d147c4bd5ec77f6f